### PR TITLE
Talk about clangd and clang-format in the docs.

### PR DESCRIPTION
**Pull request by vext01**

### CODE REVIEW

**Comment by ltratt** [comment]
> This assumes you have a `clang-format` in `$PATH`.

Hmm, I now realise that our earlier conversation only scratched the surface here. Rather than:

* burdening users with this warning
* the inevitable differences between external and ykllvm's `clang-format` s

why don't we build our own and automatically run that with `xtask fmt`? We'll win that back rather quickly, I suspect.

**Comment by vext01** [comment]
> why don't we build our own and automatically run that with xtask fmt?

I think it should tie in with `YKB_YKLLVM_BIN_DIR` (if set) like the other places in the build system?

**Comment by ltratt** [comment]
> I think it should tie in with YKB_YKLLVM_BIN_DIR (if set) like the other places in the build system?

s/should/must/ !

This should be fairly easy though, I hope. Make `xtask` depend on `ykbuild` then use `ykbuild::ykllvm_bin_dir("clang-format")` and it'll pick the right one up automatically.

File: xtask/src/main.rs [hunk 1]
```diff
@@ -25,13 +26,19 @@ fn clang_format() {
         if let Some(ext) = entry.path().extension() {
             match ext.to_str().unwrap() {
                 "h" | "c" | "cpp" | "cc" => {
-                    let res = Command::new("clang-format")
+                    let mut clang_format = ykllvm_bin_dir();
```
Comment (ltratt) [inline xtask/src/main.rs:29]: It's easier to do `llvm_bin("clang-format")`. [This will also `panic` if it can't find the binary.]

Comment (vext01) [reply xtask/src/main.rs:29]: Ohhhh.

I was confused by:

> then use ykbuild::ykllvm_bin_dir("clang-format") and it'll pick the right one up automatically.

which didn't have the right args!

Comment (ltratt) [reply xtask/src/main.rs:29]: Sorry!

Comment (vext01) [reply xtask/src/main.rs]: Try [509747c](https://github.com/ykjit/yk/pull/693/commits/509747cbe7fd82c7093e423e214481775a035117)

**Comment by ltratt** [comment]
Please rebase + squash.

**Comment by vext01** [comment]
Just realised that we now need to update the earlier doc change! One moment!

**Comment by vext01** [comment]
Does 39239e9366c3b8405d6fd2c87a5d73f84dc9e18d work for you?

**Comment by ltratt** [comment]
Please squash and rebase.

**Comment by vext01** [comment]
That was fiddly because the doc file I edited has gone in the meantime.

Arguably the `clangd` section isn't well-placed, but I'd like to move on.

**Pull request merged**